specs/protocol/rollup-node-p2p.md

```diff
@@ -111,6 +111,19 @@ The discovery process in Optimism is a pipeline of node records:
 4. Check if the record contains the `opstack` entry, verify it matches the chain ID and current or future fork number
 5. If not already connected, and not recently disconnected or put on deny-list, attempt to dial.
 
+#### Execution Layer Structure
+
+The Ethereum Node Record (ENR) for an Optimism execution node must contain the following values, identified by unique keys:
+
+- An OpStack (`opel` field) L2 network identifier.
+
+The EL uses a different key from the CL in order to stop EL and CL nodes from connecting to each other.
+
+The `opel` value is encoded as a single RLP `bytes` value, the concatenation of:
+
+- chain ID (`unsigned varint`)
+- fork ID (`unsigned varint`)
```

Comment:
do you mind using the EL standard? https://github.com/ethereum/EIPs/blob/master/EIPS/eip-2124.md
in reth we have this function for all chain specs https://github.com/paradigmxyz/reth/blob/652be135c795e7a8518078de31f8b51e219f331e/crates/primitives/src/chain/spec.rs#L769-L774

Reply:
As implemented, fork ID/version is zero for `opstack`: https://github.com/ethereum-optimism/optimism/blob/ba174f4d5f4020ff16298fefd86b55a29d4724a9/op-node/p2p/discovery.go#L79-L82

We can consider setting version to non-zero with Fjord

Reply:
@sebastianst @trianglesphere discv5 uses a list of fork IDs for the `"eth"` field. doing the same thing for `"opel"` would make sense. cc @mattsse <https://github.com/paradigmxyz/reth/pull/7900>. from specs <https://github.com/ethereum/devp2p/blob/master/enr-entries/eth.md#entry-format>